apiserver/common/model/modelstatus.go

```diff
@@ -158,6 +146,21 @@ func (c *ModelStatusAPI) modelStatus(ctx context.Context, tag string) (params.Mo
 	if err != nil {
 		return status, errors.Trace(err)
 	}
+	machineNames, err := machineService.AllMachineNames(ctx)
+	if err != nil {
+		return status, errors.Trace(err)
+	}
+
+	hostedMachineCount := 0
+	for _, machineName := range machineNames {
+		isController, err := machineService.IsMachineController(ctx, machineName)
```

Comment:
Instead of fetching this in a for loop, we can add a new machine service method to count the hostedMachineCount from the machine table.

Reply:
Or giving the ModelMachineInfo() is already terribally written, we can just get the count from there. Then we will refactor ModelMachineInfo together in the future.